### PR TITLE
Perf/fuse mamba state scatter mtp verify

### PR DESCRIPTION
**Pull request by xutizhou**
## Summary
This PR introduces a fused Triton kernel for Mamba state scatter operations in MTP (Multi-Token Prediction) verify stage, reducing kernel launch overhead and improving throughput.
## Changes
- **New fused kernel**: `fused_mamba_state_scatter_with_mask` in `mamba_state_scatter_triton.py`
  - Fuses multiple `index_elementwise_kernel` calls into a single Triton kernel
  - Handles masking internally, avoiding separate `nonzero()` and `index_select()` operations
  - Reduces kernel launch count by ~25% (253 → 189 in profile)
- **Updated backend**: `hybrid_linear_attn_backend.py`
  - Uses the new fused kernel in `update_mamba_state_after_mtp_verify()`
- **Unit tests**: `test_mamba_state_scatter_triton.py`
  - Correctness test comparing fused kernel vs reference implementation
  - Optional performance benchmark
### E2E Throughput (Qwen3-Next-80B-A3B, 8xH20, bs=48, speculative_num_steps=4)
| Metric | Baseline | Optimized | Improvement |
|--------|----------|-----------|-------------|
| Output throughput (tok/s) | 1830.60 | 1901.81 | **+3.9%** |
| Accept rate | 0.74 | 0.74 | No change |
### Profile Analysis
| Metric | Baseline | Optimized |
|--------|----------|-----------|
| `index_elementwise_kernel` calls | 253 | 189 (-25%) |
| `fused_mamba_state_scatter` calls | 0 | 32 |
## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review Process

1. Ping Merge Oncalls to start the PR flow. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - `/tag-run-ci-label`, `/rerun-failed-ci`, `/tag-and-rerun-ci`
4. After green CI and required approvals, ask Merge Oncalls to merge.
## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review Process

1. Ping Merge Oncalls to start the PR flow. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - `/tag-run-ci-label`, `/rerun-failed-ci`, `/tag-and-rerun-ci`
4. After green CI and required approvals, ask Merge Oncalls to merge.

### CODE REVIEW

File: python/sglang/srt/layers/attention/mamba/mamba_state_scatter_triton.py [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+Fused Triton kernel for Mamba state scatter operations.
+
+This kernel replaces the expensive advanced indexing operations in
+`update_mamba_state_after_mtp_verify` with a single fused gather-scatter kernel,
+avoiding multiple `index_elementwise_kernel` launches.
+"""
+
+import torch
+import triton
+import triton.language as tl
```
Comment (yizhang2077) [inline python/sglang/srt/layers/attention/mamba/mamba_state_scatter_triton.py:11]: could we move this file under the `mamba` folder?

Comment (xutizhou) [reply python/sglang/srt/layers/attention/mamba/mamba_state_scatter_triton.py:11]: done

**Review by yizhang2077 — APPROVED** [review]

**Review by yizhang2077 — APPROVED** [review]
